src/frontend/src/pages/FlowPage/components/flowSidebarComponent/components/sidebarDraggableComponent/index.tsx

```diff
@@ -0,0 +1,220 @@
+import { Badge } from "@/components/ui/badge";
+import { Button } from "@/components/ui/button";
+import useDeleteFlow from "@/hooks/flows/use-delete-flow";
+import { useAddComponent } from "@/hooks/useAddComponent";
+import { DragEventHandler, forwardRef, useRef, useState } from "react";
+import IconComponent, {
+  ForwardedIconComponent,
+} from "../../../../../../components/genericIconComponent";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+} from "../../../../../../components/ui/select-custom";
+import { useDarkStore } from "../../../../../../stores/darkStore";
+import useFlowsManagerStore from "../../../../../../stores/flowsManagerStore";
+import { APIClassType } from "../../../../../../types/api";
+import {
+  createFlowComponent,
+  downloadNode,
+  getNodeId,
+} from "../../../../../../utils/reactflowUtils";
+import { cn, removeCountFromString } from "../../../../../../utils/utils";
+
+export const SidebarDraggableComponent = forwardRef(
+  (
+    {
+      sectionName,
+      display_name,
+      icon,
+      itemName,
+      error,
+      color,
+      onDragStart,
+      apiClass,
+      official,
+      beta,
+      legacy,
+    }: {
+      sectionName: string;
+      apiClass: APIClassType;
+      icon: string;
+      display_name: string;
+      itemName: string;
+      error: boolean;
+      color: string;
+      onDragStart: DragEventHandler<HTMLDivElement>;
+      official: boolean;
+      beta: boolean;
+      legacy: boolean;
+    },
+    ref,
+  ) => {
+    const [open, setOpen] = useState(false);
+    const { deleteFlow } = useDeleteFlow();
+    const flows = useFlowsManagerStore((state) => state.flows);
+    const addComponent = useAddComponent();
+
+    const version = useDarkStore((state) => state.version);
+    const [cursorPos, setCursorPos] = useState({ x: 0, y: 0 });
+    const popoverRef = useRef<HTMLDivElement>(null);
+
+    const handlePointerDown = (e) => {
+      if (!open) {
+        const rect = popoverRef.current?.getBoundingClientRect() ?? {
+          left: 0,
+          top: 0,
+        };
+        setCursorPos({ x: e.clientX - rect.left, y: e.clientY - rect.top });
+      }
+    };
+
+    function handleSelectChange(value: string) {
+      switch (value) {
```

Comment:
what do you think if we add a default to this switch?

Reply:
the default is not needed since the select only have 2 values, both already specified... What would be the reason on that case?